File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        unordered_map<int,int> map;

       for (vector<int>::iterator it = nums.begin(); it != nums.end(); ++it) {
        if (map.find(*it) == map.end()) {
            map[*it] = 0;
        } else {
            int num = map[*it];
            map[*it] = num + 1;
            if (num > nums.size()/2 + 1) break;
        }
    }
        int max = 0;
        int element;

        for (auto i : map) {
            if (i.second >= max)  {
                max  = i.second; 
                element = i.first;
            }
        }
        return element;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        count = 0
        element = ""
        for item in nums:
            if count == 0:
                element = item
            
            if item == element:
                count += 1
            else:
                count -= 1
        
        return element